### ISSUE TITLE
Container couldn't be started

### ISSUE THREAD

**Issue by GilDarmon** [microsoft/sample-app-aoai-chatGPT#1175]
**Describe the bug**
Container couldn't be started

**To Reproduce**
Deploy as web app from Azure AI Studio

**Expected behavior**
Chat

**Screenshots**
![Image](https://github.com/user-attachments/assets/4b0efedb-a6d9-4a9d-a1d7-ce5512030c30)

**Configuration: Please provide the following**
 - Azure OpenAI model name and version (e.g. 'gpt-35-turbo-16k, version 0613') - gpt-35-turbo-16k
 - Is chat history enabled? Yes 
 - Are you using data? If so, what data source? (e.g. Azure AI Search, Azure CosmosDB Mongo vCore, etc)
- Verify the startup command and runtime configuration by showing the output of the following az CLI command:
```
az webapp show --name <app name> --resource-group <resource group name> --query "{startupCommand: siteConfig.appCommandLine, runtime: siteConfig.linuxFxVersion}"
```

**Logs**

Container gdarmon_0_32a627ef couldn't be started: Logs = _____
/ _ \ _________* _________ ____
/ /_\ \\___ / | \_ __ \_/ __ \
/ | \/ /| | /| | \/\ ___/
\____|__ /_____ \____/ |__| \___ >
\/ \/ \/
A P P S E R V I C E O N L I N U X
Documenta* http://ak*
Python 3.11.10
Note: Any data outside '/home' is not persisted
Starting OpenBSD Secure Shell server: sshd.
WEBSITES_* is not set to true.
Site's appComman* python3 -m gunicorn app:app
Launching oryx with: create-sc* -appPath /home/sit* -output /opt/star* -virtualE* antenv -defaultA* /opt/defa* -userStar* 'python3 -m gunicorn app:app'
Found build manifest file at '/home/si* Deseriali* it...
Build Operation ID: 12abcae34*
Oryx Version: 0.2.20241* Commit: 95c7a8852* ReleaseTa* 20241021.*
Output is compresse* Extractin* it...
Extractin* '/home/si* to directory '/tmp/8dd*
App path is set to '/tmp/8dd*
Writing output script to '/opt/sta*
Using packages from virtual environme* antenv located at /tmp/8dd0*
Updated PYTHONPAT* to '/opt/sta*
[2024-11-* 16:48:13 +0000] [66] [INFO] Starting gunicorn 20.1.0
[2024-11-* 16:48:13 +0000] [66] [INFO] Listening at: http://0.* (66)
[2024-11-* 16:48:13 +0000] [66] [INFO] Using worker: uvicorn.w*
[2024-11-* 16:48:13 +0000] [67] [INFO] Booting worker with pid: 67
[2024-11-* 16:48:13 +0000] [68] [INFO] Booting worker with pid: 68
[2024-11-* 16:48:13 +0000] [69] [INFO] Booting worker with pid: 69
/tmp/8dd0* UserWarni* A custom validator is returning a value other than `self`.
Returning anything other than `self` from a top level model validator isn't supported when validatin* via `__init__*
See the `model_va* docs (https://* for more details.
warnings.*
/tmp/8dd0* UserWarni* A custom validator is returning a value other than `self`.
Returning anything other than `self` from a top level model validator isn't supported when validatin* via `__init__*
See the `model_va* docs (https://* for more details.
warnings.*
/tmp/8dd0* UserWarni* A custom validator is returning a value other than `self`.
Returning anything other than `self` from a top level model validator isn't supported when validatin* via `__init__*
See the `model_va* docs (https://* for more details.
warnings.*
WARNING:r* datasourc* configura* found in the environme* -- calls will be made to Azure OpenAI without grounding data.
WARNING:r* 'missing'* 'loc': ('service* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://* {'type': 'missing'* 'loc': ('index',* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://*
WARNING:r* datasourc* configura* found in the environme* -- calls will be made to Azure OpenAI without grounding data.
WARNING:r* 'missing'* 'loc': ('service* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://* {'type': 'missing'* 'loc': ('index',* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://*
WARNING:r* datasourc* configura* found in the environme* -- calls will be made to Azure OpenAI without grounding data.
WARNING:r* 'missing'* 'loc': ('service* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://* {'type': 'missing'* 'loc': ('index',* 'msg': 'Field required'* 'input': {'top_k': '5', 'strictne* '3', 'enable_i* 'false', 'use_sema* 'false'}, 'url': 'https://*
[2024-11-* 16:48:47 +0000] [68] [ERROR] Exception in worker process
Traceback (most recent call last):
File "/tmp/8dd* line 589, in spawn_wor*
worker.in*
File "/tmp/8dd* line 66, in init_proc*
super(Uvi* self).ini*
File "/tmp/8dd* line 134, in init_proc*
self.load*
File "/tmp/8dd* line 146, in load_wsgi
self.wsgi = self.app.*
^^^^^^^^^*
File "/tmp/8dd* line 67, in wsgi
self.call* = self.load*
^^^^^^^^^*
File "/tmp/8dd* line 58, in load
return self.load*
^^^^^^^^^*
File "/tmp/8dd* line 48, in load_wsgi*
return util.impo*
^^^^^^^^^*
File "/tmp/8dd* line 359, in import_ap*
mod = importlib*
^^^^^^^^^*
File "/opt/pyt* line 126, in import_mo*
return _bootstra* package, level)
^^^^^^^^^*
File "
Container gdarmon_0_32a627ef didn't respond to HTTP pings on port: 8000, failing site start. See container logs for debugging.

1. If the application deployment is failing, please share the deployment logs using the following az CLI command:
```
az webapp log deployment show --name gdarmon --resource-group gdarmon-8027-rg
 az webapp log deployment show --name gdarmon --resource-group gdarmon-8027-rg
/usr/lib64/az/lib/python3.9/site-packages/paramiko/pkey.py:100: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "cipher": algorithms.TripleDES,
/usr/lib64/az/lib/python3.9/site-packages/paramiko/transport.py:259: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "class": algorithms.TripleDES,
[
  {
    "details_url": null,
    "id": "f509ac08-91e7-43b3-b6f9-1d654ad7cfa1",
    "log_time": "2024-11-14T16:32:23.9313793Z",
    "message": "Updating submodules.",
    "type": 0
  },
  {
    "details_url": null,
    "id": "958f6dc5-4517-4f3e-9d4f-a3682f35edb8",
    "log_time": "2024-11-14T16:32:25.1084647Z",
    "message": "Preparing deployment for commit id '771429030b'.",
    "type": 0
  },
  {
    "details_url": null,
    "id": "f4d4a7b0-e43e-42d8-b26e-1700306b9121",
    "log_time": "2024-11-14T16:32:25.4305114Z",
    "message": "PreDeployment: context.CleanOutputPath False",
    "type": 0
  },
  {
    "details_url": null,
    "id": "01118d4a-b235-4c4a-a80a-dfcc4e2dc2cd",
    "log_time": "2024-11-14T16:32:25.5593252Z",
    "message": "PreDeployment: context.OutputPath /home/site/wwwroot",
    "type": 0
  },
  {
    "details_url": null,
    "id": "e7193f10-33d8-4618-926d-7e9134faae07",
    "log_time": "2024-11-14T16:32:25.7218303Z",
    "message": "Repository path is /home/site/repository",
    "type": 0
  },
  {
    "details_url": "https://gdarmon.scm.azurewebsites.net/api/deployments/771429030bbcf210a57478d692fd8af06a3d5e01/log/55a344b3-41db-4a7c-a11b-2b9299d11811",
    "id": "55a344b3-41db-4a7c-a11b-2b9299d11811",
    "log_time": "2024-11-14T16:32:25.892613Z",
    "message": "Running oryx build...",
    "type": 0
  },
  {
    "details_url": null,
    "id": "9237ddd0-732a-46eb-b0f2-6cc82a05cc48",
    "log_time": "2024-11-14T16:34:51.476848Z",
    "message": "Running post deployment command(s)...",
    "type": 0
  },
  {
    "details_url": null,
    "id": "7e4de902-feb2-485d-829e-dfdac4e4a4d2",
    "log_time": "2024-11-14T16:34:51.5802879Z",
    "message": "",
    "type": 0
  },
  {
    "details_url": null,
    "id": "bdb4e159-218d-4a8e-a701-90752a836f36",
    "log_time": "2024-11-14T16:34:51.6316443Z",
    "message": "Generating summary of Oryx build",
    "type": 0
  },
  {
    "details_url": null,
    "id": "7518c069-d646-4e47-8983-63ffb4a6bc87",
    "log_time": "2024-11-14T16:34:51.7380543Z",
    "message": "Parsing the build logs",
    "type": 0
  },
  {
    "details_url": null,
    "id": "03064e98-a2f5-4c82-8e4b-337851783bfb",
    "log_time": "2024-11-14T16:34:51.8503698Z",
    "message": "Found 0 issue(s)",
    "type": 0
  },
  {
    "details_url": null,
    "id": "36dafd8f-0e9c-435f-9a9e-9a0dd195f14d",
    "log_time": "2024-11-14T16:34:51.9275048Z",
    "message": "",
    "type": 0
  },
  {
    "details_url": null,
    "id": "2cca046f-620c-43eb-8b08-da68adef4330",
    "log_time": "2024-11-14T16:34:51.9878067Z",
    "message": "Build Summary :",
    "type": 0
  },
  {
    "details_url": null,
    "id": "9cc33435-6a62-4b05-a456-8478a94266f9",
    "log_time": "2024-11-14T16:34:52.0463332Z",
    "message": "===============",
    "type": 0
  },
  {
    "details_url": null,
    "id": "82cd4e66-6fa4-4d97-b632-f781d8722a95",
    "log_time": "2024-11-14T16:34:52.1243816Z",
    "message": "Errors (0)",
    "type": 0
  },
  {
    "details_url": null,
    "id": "90435922-9578-455f-a018-78cc42fd8d83",
    "log_time": "2024-11-14T16:34:52.1798611Z",
    "message": "Warnings (0)",
    "type": 0
  },
  {
    "details_url": null,
    "id": "80813c90-add1-4480-9ee7-b58d860f4014",
    "log_time": "2024-11-14T16:34:52.2338927Z",
    "message": "",
    "type": 0
  },
  {
    "details_url": null,
    "id": "795f9e60-a3c5-4656-ae1c-de34cedb099d",
    "log_time": "2024-11-14T16:34:52.3599448Z",
    "message": "Triggering recycle (preview mode disabled).",
    "type": 0
  },
  {
    "details_url": null,
    "id": "2d3aa8e2-04c4-4cac-8d61-b72c6db37109",
    "log_time": "2024-11-14T16:34:52.4643461Z",
    "message": "Deployment successful. deployer = GitHub deploymentPath = ",
    "type": 0
  }
]
gill [ ~ ]$ 
```

2. If the application is crashing after deployment, please share the application logs using the following az CLI command:
```
az webapp log tail --name gdarmon --resource-group gdarmon-8027-rg
```

**Additional context**

![Image](https://github.com/user-attachments/assets/8bc0d544-f515-4f69-8e08-fd0a839c67a1)

**Comment by GilDarmon** [comment]
az webapp show --name gdarmon --resource-group gdarmon-8027-rg --query "{startupCommand: siteConfig.appCommandLine, runtime: siteConfig.linuxFxVersion}"

{
  "runtime": "PYTHON|3.11",
  "startupCommand": "python3 -m gunicorn app:app"
}

**Comment by wheatsnackbread** [comment]
Facing the exact same issue, but with the deploy button from Azure AI Studio.

**Comment by Kale-S** [comment]
Facing the exact same issue, but with the deploy button from Azure AI Studio.

**Comment by divergent-synapse** [comment]
Same here. Deploy web app from Azure AI Studio.

**Comment by wheatsnackbread** [comment]
A solution that worked for me was updating the requirements.txt to include the following:
`Flask[async]==3.0.3` 

You can do this by forking this repository and then updating requirements.txt in the root folder. Commit all changes.

Then, redeploy the app with these changes.
1. In Azure Portal, navigate to your web app instance. 
2. Go to Deployment > Deployment Center.
3. Under source, hit Disconnect. 
4. Add a new source, selecting the repo from your GitHub.
5. Wait ~15 minutes for it to fully redeploy. You may want to hit the sync button in Deployment Center and 'Restart' under Web App a few times for good measure.

If you run into errors with searching through the Azure AI Search index ("chat history can't be saved at this time" / managed identity issues) this is a known problem as well. Simply go to your AI search instance and copy the API key from Settings > Keys, then paste it under Environment Variables > AZURE_SEARCH_KEY. 

Someone needs to write a blog or LinkedIn post or something about how to resolve issues with this app lol. There's recurring issues with conversation history (Cosmos DB failing to deploy / overcapacity in busy regions) as well—Canada Central seems to work better than East US or East US 2 as of the past two days.

**Comment by GdarmonAbra** [comment]
New Version 
updating the requirements.txt to include the following:
Flask[async]==3.0.3
fixed the issue, Thanks!

**Comment by sarah-widder** [comment]
Fixed by #1174
